pandas/tests/io/pytables/test_store.py

```diff
@@ -4528,10 +4527,8 @@ def test_read_hdf_iterator(self, setup_path):
             df.to_hdf(path, "df", mode="w", format="t")
             direct = read_hdf(path, "df")
             iterator = read_hdf(path, "df", iterator=True)
-            assert isinstance(iterator, TableIterator)
             indirect = next(iterator.__iter__())
             tm.assert_frame_equal(direct, indirect)
-            iterator.store.close()
```

Comment:
I think its a problem not closing the iterator; the fact that you had to change this is suspicious

Reply:
`iter_result` _does_  close the store